src/mlpack/core/optimizers/sdp/lrsdp_function_impl.hpp

```diff
@@ -189,5 +187,40 @@ void AugLagrangianFunction<LRSDPFunction>::Gradient(
   gradient = 2 * s * coordinates;
 }
 
+// Template specializations for function and gradient evaluation.
+// Note that C++ does not allow partial specialization of class members,
+// so we have to go about this in a somewhat round-about way.
+template <>
+inline double AugLagrangianFunction<LRSDPFunction<SDP<arma::sp_mat>>>::Evaluate(
+    const arma::mat& coordinates) const
+{
+  return EvaluateImpl(function, coordinates, lambda, sigma);
+}
```

Comment:
Two questions:

  1. How does this avoid the no partial specialization allowed for class members (see comment)
  2. Doesn't this require us to change the type signature on `AugLagrangianFunction`?